### ISSUE TITLE
Problem with regression using l2_loss

### ISSUE THREAD

**Issue by rmldj** [dmlc/cxxnet#120]
I have problems with doing regression using the l2_loss layer.

In my netconfig section I have the following:

```
layer[4->5] = fullc
  nhidden = 1
layer[5->5] = l2_loss
 netconfig=end

input_shape = 1,1,11
```

I try to evaluate the fit using rmse but I get only nan's. I traced back the problem to the fact that the ```MetricBase``` class in ```metric.h``` gets nan's as predicted values. I put in the following diagnostic code there:

```
  virtual void AddEval(const mshadow::Tensor<cpu,2> &predscore, const LabelRecord& labels) {
    for (index_t i = 0; i < predscore.size(0); ++ i) {
      sum_metric += CalcMetric(predscore[i], labels.label[i]);
      cnt_inst+= 1;
    } printf("addeval label %f predscore %f \n",labels.label[0][0],predscore[0][0]);
  }
```

The labels (regression targets) are correct but ```predscore``` gives -nan.

The data is input using my custom npz reader which works however without any problems for classification problems (and here the regression targets are also read in correctly).

In fact if I change the last layer to softmax (with 8 nodes for this dataset - the wine quality example dataset) then I see normal numerical predictions in ```MetricBase::AddEval```.

Did anyone succeed in running some regression with l2_loss? I would be grateful for an example config file and data...

**Comment by winstywang** [comment]
l2_loss usually need a much smaller learning rate. Try 1e-6 or 1e-7 first.

**Comment by rmldj** [comment]
Thanks!! 
I just also figured out that this is a problem when trying to find the first output from the gradient.
Decreasing the learning rate to something like 0.001 already gave finite numbers...
Sorry for the trouble..
